src/mat/4C_mat_fourieriso.cpp

```diff
@@ -1,137 +0,0 @@
-void Mat::FourierIso::pack(Core::Communication::PackBuffer& data) const
-{
-  // pack type of this instance of ParObject
-  int type = unique_par_object_id();
-  add_to_pack(data, type);
-
-  // matid
-  int matid = -1;
-  if (params_ != nullptr) matid = params_->id();  // in case we are in post-process mode
-  add_to_pack(data, matid);
-}
-
-/*----------------------------------------------------------------------*
- |  Unpack                                        (public)  bborn 04/09 |
- *----------------------------------------------------------------------*/
-void Mat::FourierIso::unpack(Core::Communication::UnpackBuffer& buffer)
-{
-  Core::Communication::extract_and_assert_id(buffer, unique_par_object_id());
-
-  // matid
-  int matid;
-  extract_from_pack(buffer, matid);
-  params_ = nullptr;
-  if (Global::Problem::instance()->materials() != nullptr)
-    if (Global::Problem::instance()->materials()->num() != 0)
-    {
-      const int probinst = Global::Problem::instance()->materials()->get_read_from_problem();
-      Core::Mat::PAR::Parameter* mat =
-          Global::Problem::instance(probinst)->materials()->parameter_by_id(matid);
-      if (mat->type() == material_type())
-        params_ = static_cast<Mat::PAR::FourierIso*>(mat);
-      else
-        FOUR_C_THROW("Type of parameter material %d does not fit to calling type %d", mat->type(),
-            material_type());
-    }
-}
-
-/*----------------------------------------------------------------------*
- |  calculate for 1D                                         dano 09/09 |
- *----------------------------------------------------------------------*/
-void Mat::FourierIso::evaluate(const Core::LinAlg::Matrix<1, 1>& gradtemp,
-    Core::LinAlg::Matrix<1, 1>& cmat, Core::LinAlg::Matrix<1, 1>& heatflux) const
-{
-  // conductivity tensor
-  cmat(0, 0) = params_->conduct_;
-
-  // heatflux
-  heatflux.multiply_nn(cmat, gradtemp);
-}
-
-/*----------------------------------------------------------------------*
- |  calculate for 2D                                         dano 09/09 |
- *----------------------------------------------------------------------*/
-void Mat::FourierIso::evaluate(const Core::LinAlg::Matrix<2, 1>& gradtemp,
-    Core::LinAlg::Matrix<2, 2>& cmat, Core::LinAlg::Matrix<2, 1>& heatflux) const
-{
-  // conductivity tensor
-  cmat.clear();
-  for (int i = 0; i < 2; ++i) cmat(i, i) = params_->conduct_;
-
-  // heatflux
-  heatflux.multiply_nn(cmat, gradtemp);
-}
-
-/*----------------------------------------------------------------------*
- |  calculate for 3D                                         dano 09/09 |
- *----------------------------------------------------------------------*/
-void Mat::FourierIso::evaluate(const Core::LinAlg::Matrix<3, 1>& gradtemp,
-    Core::LinAlg::Matrix<3, 3>& cmat, Core::LinAlg::Matrix<3, 1>& heatflux) const
-{
-  // conductivity tensor
-  cmat.clear();
-  for (int i = 0; i < 3; ++i) cmat(i, i) = params_->conduct_;
-
```

Comment:
E.g. the conductivity value was given as scalar in the input file, but internally the tensor is still created by setting the values onto the diagonal, so the matrix-vector is done anyway.